src/Mvc/Mvc.Core/src/ProducesResponseTypeAttribute.cs

```diff
@@ -13,6 +15,8 @@ namespace Microsoft.AspNetCore.Mvc
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
     public class ProducesResponseTypeAttribute : Attribute, IApiResponseMetadataProvider
     {
+        private readonly MediaTypeCollection _contentTypes = new();
```

Comment:
This is nitpicky but we could avoid this allocation in the other code path if we declare this as nullable.